test/sql/bgw_scheduler.sql

```diff
@@ -0,0 +1,121 @@
+ * and the scheduler for single in pg_stat_activity
+ * but single_2 shouldn't have a scheduler because ext not created yet 
+ */
+SELECT * FROM worker_counts;
+
+/*Now create the extension in single_2*/
+CREATE EXTENSION timescaledb CASCADE;
+select pg_sleep(0.1);
+
+/* and we should now have a scheduler for single_2*/
+SELECT * FROM worker_counts;
+
+DROP DATABASE single;
+
+/* Now the db_scheduler for single should have disappeared*/
+SELECT * FROM worker_counts;
+
+/*now let's restart the scheduler and make sure our backend_start changed */
+SELECT backend_start as orig_backend_start 
+FROM pg_stat_activity 
+WHERE application_name = 'Timescale BGW Scheduler Entrypoint' 
+AND datname = 'single_2' \gset
+/* we'll do this in a txn so that we can see that the worker locks on our txn before continuing*/
+BEGIN;
+SELECT _timescaledb_internal.restart_background_workers();
+select pg_sleep(0.1);
+
+SELECT * FROM worker_counts;
+
+SELECT (backend_start > :'orig_backend_start'::timestamptz) backend_start_changed, 
+(wait_event = 'virtualxid') wait_event_changed
+FROM pg_stat_activity 
+WHERE application_name = 'Timescale BGW Scheduler Entrypoint' 
+AND datname = 'single_2';
+COMMIT;
+
+SELECT * FROM worker_counts;
+
+SELECT (wait_event IS DISTINCT FROM 'virtualxid') wait_event_changed
+FROM pg_stat_activity 
+WHERE application_name = 'Timescale BGW Scheduler Entrypoint' 
+AND datname = 'single_2';
+
+
+/*test stop*/
+SELECT _timescaledb_internal.stop_background_workers();
+SELECT * FROM worker_counts;
+/*make sure it doesn't break if we stop twice in a row*/
+SELECT _timescaledb_internal.stop_background_workers();
+SELECT * FROM worker_counts;
+
+/*test start*/
+SELECT _timescaledb_internal.start_background_workers();
+select pg_sleep(0.1);
+
+SELECT * FROM worker_counts;
+
+/*make sure start is idempotent*/
+SELECT backend_start as orig_backend_start 
+FROM pg_stat_activity 
+WHERE application_name = 'Timescale BGW Scheduler Entrypoint' 
+AND datname = 'single_2' \gset
+
+SELECT _timescaledb_internal.start_background_workers();
+select pg_sleep(0.1);
+
+SELECT (backend_start = :'orig_backend_start'::timestamptz) backend_start_unchanged
+FROM pg_stat_activity 
+WHERE application_name = 'Timescale BGW Scheduler Entrypoint' 
+AND datname = 'single_2';
+
+/*Make sure restart works from stopped worker state*/
+SELECT _timescaledb_internal.stop_background_workers();
+SELECT * FROM worker_counts;
+SELECT _timescaledb_internal.restart_background_workers();
+select pg_sleep(0.1);
+SELECT * FROM worker_counts;
+
+/*Make sure drop extension statement restarts the worker and on rollback it keeps running*/
+
+/*now let's restart the scheduler and make sure our backend_start changed */
+SELECT backend_start as orig_backend_start 
+FROM pg_stat_activity 
+WHERE application_name = 'Timescale BGW Scheduler Entrypoint' 
+AND datname = 'single_2' \gset
+
+BEGIN;
+DROP EXTENSION timescaledb;
+select pg_sleep(0.1);
```

Comment:
Not sure what behavior you're trying to test here. If we drop the extension within a transaction that gets rolled back... then this shouldn't even be visible to the scheduler / launcher?